lnd_test.go

```diff
@@ -1407,6 +1407,329 @@ func testMultiHopPayments(net *networkHarness, t *harnessTest) {
 	}
 }
 
+// testMultiHopPaymentsToRoute
+func testMultiHopPaymentsToRoute(net *networkHarness, t *harnessTest) {
```

Comment:
I believe that the functionality which is introduced in this `PR` might be tested with less efforts, basically we could create one channel between `Alice` and `Bob` and send payment to the unknown receiver, by this we will be `100%` sure that daemon took our route instead of generating path automatically. Also keeping the test smaller helps as maintain it latter.